### PR TITLE
Add generic API lookup plugin

### PR DESCRIPTION
**Pull request by Tompage1994**
### What does this PR do?
Adds a generic API lookup plugin to the collection similar to https://docs.ansible.com/ansible/latest/collections/awx/awx/controller_api_lookup.html

The code was lifted from the awx.awx collection and modified as appropriate

### How should this be tested?
I've tested against api, _ui and pulp type endpoints and have ensured working with local and keycloak users for each.

I used this playbook which gave a good overview of the params too.
```
- hosts: localhost
  gather_facts: false
  tasks:
    - debug:
        msg: "{{ lookup('redhat_cop.ah_configuration.ah_api', 'ee_namespaces', host='https://ah.local', username='admin', password='password', verify_ssl=false, return_all=true) | map(attribute='name') | list }}"
    - debug:
        msg: "{{ lookup('redhat_cop.ah_configuration.ah_api', 'users', host='https://ah.local', username='admin', password='password', verify_ssl=false, query_params={'username__startswith': 'a'}) }}"
    - debug:
        msg: "{{ lookup('redhat_cop.ah_configuration.ah_api', 'namespaces', host='https://ah.local', username='admin', password='password', verify_ssl=false, query_params={'company': 'Community Test'}) }}"
    - debug:
        msg: "Users: {{ lookup('redhat_cop.ah_configuration.ah_api', 'ee_namespaces', host='https://ah.local', username='ansible', password='password', verify_ssl=false, return_all=true) | map(attribute='name') | list }}"
    - debug:
        msg: "{{ lookup('redhat_cop.ah_configuration.ah_api', 'users', host='https://ah.local', username='ansible', password='password', verify_ssl=false, return_all=true) | map(attribute='username') | list }}"
    - debug:
        msg: "{{ lookup('redhat_cop.ah_configuration.ah_api', 'namespaces', host='https://ah.local', username='ansible', password='password', verify_ssl=false, return_all=true) | map(attribute='name') | list }}"
```

### Is there a relevant Issue open for this?
N/A

### Other Relevant info, PRs, etc.
A customer pointed out there was no easy way to get info from the AH before doing additional stuff without going through the API and having to deal with the auth etc. This will solve that.

### CODE REVIEW

File: plugins/lookup/ah_api.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+- name: Report the usernames of all users
+  debug:
+    msg: "Users: {{ query('redhat_cop.ah_configuration.ah_api', 'users', return_all=true) | map(attribute='username') | list }}"
+
+- name: List all collection namespaces by the devops team
+  debug:
+    msg: "{{ lookup('redhat_cop.ah_configuration.ah_api', 'namespaces', host='https://ah.example.com', username='ansible',
+              password='Passw0rd123', verify_ssl=false, query_params={'company': 'Devops'}) }}"
+
+- name: Get the list of tags for my_ee
```
Comment (sean-m-sullivan) [inline plugins/lookup/ah_api.py:90]: I think there needs to be some more documentation around these two and the options that are required, I know you put in notes above, but think its slightly confusing.

**Review by sean-m-sullivan — CHANGES_REQUESTED** [review]
LGTM I just think needs some additional documentation, This is awesome and powerful!

File: plugins/lookup/ah_api.py (hunk 1 above)
Comment (Tompage1994) [reply plugins/lookup/ah_api.py:90]: I can add it to the notes section, but I'm not entirely sure how better to document it other than to add to the notes to look at the examples.

Comment (Tompage1994) [reply plugins/lookup/ah_api.py:90]: Pushed an extra note and perhaps made the choices more (or less?) clear

Comment (sean-m-sullivan) [reply plugins/lookup/ah_api.py:90]: It's been clear to me, Just worried that once people discover this feature they are going to create issues 'not' understanding it, this is good as is, we can figure out more documentation later, maybe a readme spelling it out for people not used to lookup plugins :)

**Review by sean-m-sullivan — APPROVED** [review]

**Pull request merged**